### PR TITLE
Implemented user registration

### PR DESCRIPTION
**Pull request by dincciftci**
This allows for a new reddit account to be created. I tested this, but it's not possible to do unit tests.

I had to edit HttpRestClient.java as such--because there is no User variable for an unregistered person, there is no cookie, and I encountered errors in the given line. As far as I understand, this should not change the behavior any existing functions.

### CODE REVIEW

**Comment by dincciftci** [comment]
I updated the implemented methods file, and fixed a few other small things that were causing problems when using the library. Let me know if there is anything in particular that you don't like.

File: src/main/java/com/github/jreddit/action/ProfileActions.java [hunk 1]
```diff
@@ -140,6 +141,9 @@ public UserInfo getUserInformation() throws ActionFailedException {
     public UserInfo about(String username) throws ActionFailedException {
 
         // Send GET request to get the account overview
+      
+        if(username == "[deleted]")
```
Comment (sfat) [inline src/main/java/com/github/jreddit/action/ProfileActions.java]: can you add curly brackets to the if statement? it good practice so you don't end up with weird bugs in the future

File: src/main/java/com/github/jreddit/action/ProfileActions.java [hunk 2]
```diff
@@ -162,4 +175,53 @@ public Response delete(Boolean confirm, String message, String password) throws
...
+     * @param copyPassword  should be same as newPassword (for validation)
+     * @param captcha_iden  the identifier of the CAPTCHA challenge (not necessarily required)
+     * @param captcha_sol   the user's response to the CAPTCHA challenge (required if there was a CAPTCHA challenge)
+     * @throws ActionFailedException        If the action failed
+     */
+    public Response register(String username, String email, String newPassword, String copyPassword, String captcha_iden, String captcha_sol) throws ActionFailedException {
+        
+        // Format parameters
+        String params = 
+                "api_type=json"
```
Comment (sfat) [inline src/main/java/com/github/jreddit/action/ProfileActions.java]: is there a reason why this identation is like this and not in a straight line?

Comment (dincciftci) [inline src/main/java/com/github/jreddit/action/ProfileActions.java]: Yeah, sleep deprivation

File: src/main/java/com/github/jreddit/action/ProfileActions.java (hunk 1 above)
Comment (dincciftci) [inline src/main/java/com/github/jreddit/action/ProfileActions.java]: Done, also used the .equals() method to compare

**Pull request merged**